grails-app/controllers/org/broadinstitute/orsp/api/DataUseController.groovy

```diff
@@ -23,4 +43,56 @@ class DataUseController extends AuthenticatedController {
         render queryService.findDataUseRestrictions(pagination) as JSON
     }
 
+    def findRestriction() {
+        UtilityClass.registerSampleCollectionMarshaller()
+        DataUseRestriction restriction = DataUseRestriction.findById(params.id)
+        List<String> summary = consentService.getSummary(restriction)
+        Issue consent = queryService.findByKey(restriction.consentGroupKey)
+        String consentGroupName = consent.summary
+        ConsentResource consentResource = consentService.buildConsentResource(restriction, consentGroupName)
+        Collection<ConsentCollectionLink> collectionLinks = queryService.findCollectionLinksByConsentKey(restriction.consentGroupKey)
+        collectionLinks.retainAll { it.sampleCollectionId != null }
+        Set<SampleCollection> samples = collectionLinks.collect{ it.sampleCollection }
+        JSON.use(UtilityClass.SAMPLES) {
+            render([restriction    : restriction,
+                    restrictionUrl : restriction.getConsentUrl(getConsentServiceUrl()),
+                    summary        : summary,
+                    consent        : consent,
+                    samples: samples,
+                    consentResource: consentResource.toString()] as JSON)
+
+        }
+    }
+
+    def exportConsent() {
+        try {
+            DataUseRestriction restriction = DataUseRestriction.findById(params.id)
+            Collection<String> sampleCollectionIds = queryService.findAllSampleCollectionIdsForConsent(restriction.getConsentGroupKey())
+            StorageDocument dataUseLetter = null
+            List<StorageDocument> duls = queryService.getDataUseLettersForConsent(restriction.getConsentGroupKey())
+            Issue consent = queryService.findByKey(restriction.consentGroupKey)
+            if (duls && duls.size() > 0) {
+                dataUseLetter = storageProviderService.populateDocumentFileContent(duls.get(0))
+            }
+            ConsentResource resource = consentExportService.exportConsent(
+                    getUser(),
+                    restriction,
+                    sampleCollectionIds,
+                    dataUseLetter,
+                    consent.summary)
+            response.status = 200
+            render([message: "Consent successfully exported to DUOS: $resource.name"] as JSON)
+        } catch (ConsentException e) {
+            response.status = 500
+            render(status: 500, text: e as JSON, contentType: 'application/json')
+        } catch (Exception e) {
+            response.status = 500
+            render(status: 500, text: e as JSON, contentType: 'application/json')
+        }
```

Comment:
These two exception cases do the exact same thing. Should be collapsed into a single catch.